src/apps/dictionaries/redux/actions.ts

```diff
@@ -354,10 +355,10 @@ export const editDictionaryVersionAction = createActionThunk(
 );
 
 export const recursivelyAddConceptsToDictionaryAction = (
-  sourceUrl: string,
   dictionaryUrl: string,
   rawConcepts: (APIConcept | string)[],
-  bulk: boolean = false
+  bulk: boolean = false,
+  sourceUrl?: string
 ) => {
   return async (dispatch: Function, getState: Function) => {
     const concepts = rawConcepts.map(concept =>
```

Comment:
To address my previous concern, this should look like this:

```ts
    if (!!!sourceUrl && !!rawConcepts.find((c) => typeof c === "string")) {
      throw {
        message: "Cannot load string-only concepts without a source url",
      };
    }

    const concepts = rawConcepts.map((concept) =>
      typeof concept === "string"
        ? {
            id: concept,
            url: `${sourceUrl}concepts/${concept}/`,
            source_url: sourceUrl,
          }
        : concept
    );
```

Note that this throws an error if `rawConcepts` contains any strings an no sourceUrl was provided. It also adds the `source_url` property to our concepts so we can use this in the `groupBy` below.